### ISSUE TITLE
关于isa模块疑惑

### ISSUE THREAD

**Issue by swjtulinxi** [openseg-group/openseg.pytorch#35]
你好，从代码来看，通过长短两步，虽然每一步的图片的尺寸变小了，但是他的batchsize变大了啊，那么他是怎么减小了计算量和参数量的呢，非常不理解这点，请大佬解释一下

**Comment by LayneH** [comment]
Hi,

The reason is that the complexity of self-attention is quadratic with respect to the number of the pixels (i.e., size of feature map), while it is linear with respect to the batch size.

**Comment by swjtulinxi** [comment]
hi，i have a question about isa module and oc module as follows:
in isa:    self.W = nn.Sequential(
            nn.Conv2d(self.value_channels, self.out_channels, kernel_size=1, bias=False),
            ModuleHelper.BNReLU(self.out_channels, bn_type=bn_type)
        )
in oc:  self.W = nn.Conv2d(in_channels=self.value_channels, out_channels=self.out_channels,
                           kernel_size=1, stride=1, padding=0)
        nn.init.constant_(self.W.weight, 0)
        nn.init.constant_(self.W.bias, 0)

why is it different between oc  and  isa,    including  the weights of Convolution  , BN,Relu, why?

**Comment by LayneH** [comment]
Hi,

Since ISA performs two consecutive SA operations, we apply BN->ReLU between them in order to ensure the scales of the inputs of these two operations are similar. Besides, if we zero-initialize the weight of W, the input of the second SA will also be all-zeros, which hinders the learning of the second SA operation.

**Comment by swjtulinxi** [comment]
so，zero-initialize the weight of W  or  not  won't influence the result  in base oc ???

**Comment by LayneH** [comment]
Basically, this initialization does not make a huge difference for the Base-OC.

**Comment by swjtulinxi** [comment]
hi ,i have a question about the contextual module in Semantic segmentation ,such as your ISA, ppm,aspp and so on, Why I use and not use this module has no effect on the results，Does the size of batchsize affect the accuracy of the context module

**Comment by hsfzxjy** [comment]
@swjtulinxi Hi. You may provide relavant experiments settings and results.

**Comment by swjtulinxi** [comment]
感谢，我还有一个问题就是  得到的注意力权重你们是怎么可视化的啊，用gradcam？
